### DISCUSSION TITLE
hydrostatic pressure

### DISCUSSION THREAD

**Discussion by panosz** [oemof/tespy#462] [Q&A] [answered]
Is there a way to specify the height of components or connections, so that the simulation takes into account the hydrostatic pressure differences due to differences in elevation?

**Comment by fwitte** [comment, marked as answer]
Hi @panosz!

In the current state of the package: Not out of the box! What you can do is to add user defined equations to your model, that offset the pressure over a component (you could do that also with valve as pseudo change in height). It depends a little bit on what you want to do exactly.

If you need to incorporate the changes in height (and/or flow velocity) in components, it is more work. However, with the most recent changes introduced in v0.7 the framework should be extendable in that direction rather easily. If you'd like to commit some time to developing the required features yourself, you are welcome to do so, I will try to support you with that. I can give you a quick overview where to start, if you want to learn more about it. Basically, the steps are:

- register an additional variable per connection
- make tespy enable/disable height (and optionally flow velocity) as variables in your network
- modify/add respective equations to the components, which are affected by change of height or flow velocity

Best

Francesco

**Comment by panosz** [comment]
@fwitte
Thank you for your suggestions. 
However I don't feel confident in implementing the solution you propose successfully.

At the moment I am on the look of python libraries to support simple models for simulating district heating networks.

Although I have invested substantial time in familiarizing myself with tespy, I am getting more and more convinced that this library is too complicated for my purposes. 
Although there is a related example of a moderately complex district heating network in tespy [here](https://github.com/oemof/oemof-examples/tree/master/oemof_examples/tespy/district_heating) I have found that this is hard to modify and initialize successfully. Moreover, my modifications stopped working when I upgraded from version 0.6.2 to 0.7.0.

In a recent [issue](https://github.com/oemof/DHNx/issues/119), the developers of oemof's DHNx, which is a package dedicated to district heating by the same organization as the one developing tespy, seem to suggest using pandapipes for such purposes. Moreover pandapipes supports heights out of the box, so it seems that this may be a better solution.

I realize that this discussion digresses from my initial question, but I would like to know if you have any thoughts on this.

**Reply by fwitte** [reply]
Hi @panosz!

happy new year and sorry for not coming back earlier to this, it got buried in my todo-list...

I have never worked with pandapipes, but as far as I know, you can do these things with it. TESPy was never designed specifically for heat network modeling, it was just a goodie (the fact that you can model such systems) coming with the methodology. In DHNx you would optimize the layout of the heating system (in terms of routing and pipe capcities as well as heat supply) and then you could transfer those data to pandapipes. I think DHNx should even be able to handle GIS based inputs, not quite sure though. Pandapipes is to some extent also similar to tespy, but I'd expect it is more suited to deal with this kind of networks.

Best

Francesco

**Comment by panosz** [comment]
Hi, @fwitte!

Thank you for your reply. I understand now that TESPy is an overkill for such simulations and my choosing it instead of pandapipes was probably misguided. However I did so, because `district heating` was explicitly mentioned in the description of TESPy and because there is there was a basic tutorial dedicated to district heating networks.

Maybe you should add some warning in the documentation to prevent similar future misunderstandings.

Best,
Panos
